### PR TITLE
[WIP] Config options

### PR DESCRIPTION
**Pull request by webstergd**
This is a WIP pull request. Please provide feedback.

This is the start of work aimed to add configuration options. Given the Akka madness, I am posting this for feedback. Please provide feedback on ways to optimize, adjust structure, or other config options we should also break out.

### CODE REVIEW

File: src/main/scala/org/novetta/zoo/actors/RabbitConsumerActor.scala [hunk 1]
```diff
@@ -17,12 +17,13 @@ import org.novetta.zoo.types._
...
 import scala.concurrent.duration.{FiniteDuration, _}
 import org.novetta.zoo.types.WorkEncoding
+import org.novetta.zoo.util.DownloadSettings
 /**
  * @constructor This is the companion object to the RabbitConsumerActor class.
  */
 object RabbitConsumerActor {
-  def props[T: Manifest](host: HostSettings, exchange: ExchangeSettings, queue: QueueSettings, servicelist: WorkEncoding, decoder: Parsers.Parser[T]): Props = {
-    Props(new RabbitConsumerActor(host, exchange, queue, servicelist, decoder) )
+  def props[T: Manifest](config: DownloadSettings, host: HostSettings, exchange: ExchangeSettings, queue: QueueSettings, servicelist: WorkEncoding, decoder: Parsers.Parser[T]): Props = {
```
Comment (zhanif3) [inline src/main/scala/org/novetta/zoo/actors/RabbitConsumerActor.scala]: I'd reorder this, so the config is the last argument.

**Comment by zhanif3** [comment]
I'm pretty good with this, we're just passing the config object around. This is a little heavier weight than I'd really like, but for the moment, this isn't a huge ordeal, as we are not making modifications, and it should just be memory pointers.

**Comment by webstergd** [comment]
cool. I will make it current and do a final pull.

**Comment by webstergd** [comment]
Moved config locations and synced with current master. Any other requests?

**Pull request merged**